src/modules/blocks/create-accordion.js

```diff
@@ -0,0 +1,29 @@
+import { replaceElement, toBlock } from '../utils/dom-utils.js';
+
+/**
+ *
+ * @param {Document} document
+ */
+export default function createAccordion(document) {
+  Array.from(document.querySelectorAll('details')).forEach(
+    (detailsElement) => {
+      const cells = [];
+      const children = Array.from(detailsElement.childNodes);
+      children.forEach((childElement) => {
+        if (typeof childElement.innerHTML !== 'undefined') {
```

Comment:
The collapsible structure is always predictable, see:
https://github.com/bioruebe/markdown-it-collapsible

There's always a `summary` that has a `span` child (which you can ignore) and a text.
then everything under that is content.

You could do this by means of splitting the children of the details element to 2 arrays:

array one, has first child (remove the span)
array two has the rest

pass those two arrays into the `cells` param of `toBlock` and that will do it

```
// Split arry with destructuring syntax 
const [firstEl, ...rest] = Array.from(detailsElement.children)
// remove span child from firstEl
....

call `toBlock('accordion',[firstEl, rest], document)`




```

Reply:
The children array had these values below. I'll try the destructuring syntax. Thanks!

undefined
<span class="details-marker">&nbsp;</span>Click me
undefined
This is text inside a collapsible section.
undefined

Reply:
this is because you are using `childNodes`, try `children`
`childNodes` includes non elements, like text nodes, but you should not see text nodes here, unless you do? in which case use `dom-utils.js#groupWithParagraphs`

Reply:
Hi @ahmed-musallam - all set. thanks for the pointers. had to `toBlock('accordion', [[firstEl.innerHTML], rest], document)` as `toBlock() `is throwing an error if `firstEl` only.

I also had to rebase from main so you'll see more files appears modified. Is this the preferred method?